APVCV-517: Removing caching and refactoring

Abdiiir opened the pull request.

Commit:
APVCV-517: Removing caching and refactoring

FinnNielsen commented on iot-commercial-domain/src/main/java/org/dcsa/iot/commercial/domain/persistence/repository/specification/EventCacheSpecification.java:
```diff
@@ -32,53 +37,63 @@ public record EventCacheFilters(
-          jsonPath
-            .of(IoTCommercialEvent_.IOT_EVENT_TYPE_CODE)
-            .in(filters.iotEventTypeCodes.stream().map(Enum::name).toList()));
+     //   predicates.add(root.get(IoTCommercialEvent_.IOT_EVENT_TYPE_CODE).in(filters.iotEventTypeCodes.stream().map(Enum::toString).toList()));
```
Should either be made to work or removed

Abdiiir replied:
iotEventTypeCode is added again as filter now.

FinnNielsen commented on iot-commercial-service/src/main/java/org/dcsa/iot/commercial/service/unofficial/UnofficialIoTCommercialEventService.java:
```diff
@@ -0,0 +1,71 @@
+  }
+
+  @Transactional(Transactional.TxType.MANDATORY)
+  public void saveDocumentReferences(
```
should this not be private?

Abdiiir replied:
Yes, it should.

FinnNielsen reviewed (changes requested):
two minor nitpicks and you need to fix the postman tests

Commit:
APVCV-517: PR fixes

- formatted code
- added  iotEventTypeCodes as filter
- fixed misspelling in entity leading to empty pointer
- added missing mapping for DocumentReference to TO
-  updated postman tests

Abdiiir commented:
@FinnNielsen postman is updated and I have addressed the issues you pointed out. All changes is mentioned in commit message too.
